libp2p/NodeTable.cpp

```diff
@@ -453,22 +448,23 @@ void NodeTable::onPacketReceived(
                 bool expected = false;
                 auto now = chrono::steady_clock::now();
                 DEV_GUARDED(x_findNodeTimeout)
-                    m_findNodeTimeout.remove_if([&](NodeIdTimePoint const& t)
+                m_findNodeTimeout.remove_if([&](NodeIdTimePoint const& t) noexcept {
```

Comment:
I usually add braces around the statement under `DEV_GUARDED` to coerce clang-format to better indentation.
Maybe put lambda to a separate variable to simplify this.